src/LmdbCache.ts

```diff
@@ -73,8 +76,35 @@ export class LmdbCache<V = any> implements SortKeyCache<V> {
     }
   }
 
-  async put(stateCacheKey: CacheKey, value: V): Promise<void> {
-    await this.db.put(`${stateCacheKey.contractTxId}|${stateCacheKey.sortKey}`, value);
+  async put(cacheKey: CacheKey, value: V): Promise<void> {
+    await this.db.transaction(() => {
+      this.db.putSync(`${cacheKey.contractTxId}|${cacheKey.sortKey}`, value);
+
+      // Make sure there isn't too many entries for one contract
+      const numInCache = 1 + this.db.getKeysCount({
+        start: `${cacheKey.contractTxId}|${genesisSortKey}`,
+        end: `${cacheKey.contractTxId}|${cacheKey.sortKey}`
+      })
+      if (numInCache <= this.maxEntriesPerContract) {
+        // We're below the limit, finish
+        return
+      }
+
+      // Remove oldest entries, so after the final put there's minEntriesPerContract present
+      let numToRemove = numInCache - this.minEntriesPerContract
```

Comment:
I believe we can fall into negative values here? in `numValues` is lower than `minEntriesPerContract`.
Will it not cause some issues?
Also, should we simply return if numInCache <= this.minEntriesPerContract?

But again - I would leave one parameter - the max entries per contract.

Reply:
I'm assuming maxEntriesPerContract >= minEntriesPerContract.
I checked and negative values cause noop as expected, so all values are accepted.